app/controllers/document_lead_image_controller.rb

```diff
@@ -5,79 +5,6 @@ def index
     @document = Document.find_by_param(params[:document_id])
   end
 
-  def create
-    document = Document.find_by_param(params[:document_id])
-
-    unless params[:image]
-      redirect_to document_lead_image_path, alert: t("document_lead_image.index.no_file_selected")
-      return
-    end
-
-    image_uploader = ImageUploader.new(params.require(:image))
-
-    unless image_uploader.valid?
-      redirect_to document_lead_image_path, alert: {
-        "alerts" => image_uploader.errors,
-        "title" => t("document_lead_image.index.error_summary_title"),
-      }
-      return
-    end
-
-    begin
-      image = image_uploader.upload(document)
-      image.asset_manager_file_url = upload_image_to_asset_manager(image)
-    rescue GdsApi::BaseError
-      redirect_to document_lead_image_path, alert_with_description: t("document_lead_image.index.flashes.api_error")
-      return
-    end
-
-    image.save!
-    redirect_to crop_document_lead_image_path(params[:document_id], image.id, wizard: params[:wizard])
-  end
-
-  def crop
-    @document = Document.find_by_param(params[:document_id])
-    @image = @document.images.find(params[:image_id])
-  end
-
-  def update_crop
-    document = Document.find_by_param(params[:document_id])
-    image = Image.find(params[:image_id])
-
-
-    begin
-      Image.transaction do
-        image.update!(update_crop_params)
-        asset_manager_file_url = upload_image_to_asset_manager(image)
-        delete_image_from_asset_manager(image)
-        image.asset_manager_file_url = asset_manager_file_url
-        image.save!
-      end
-
-      DocumentUpdateService.update!(
-        document: document,
-        user: current_user,
-        type: "image_updated",
-        attributes_to_update: {},
-      )
-    rescue GdsApi::BaseError
-      redirect_to document_lead_image_path(document), alert_with_description: t("document_lead_image.index.flashes.api_error")
-      return
-    end
-
-    if params[:wizard].present?
-      redirect_to edit_document_lead_image_path(document, image, params.permit(:wizard))
-      return
-    end
-
-    redirect_to document_lead_image_path(document)
-  end
-
-  def edit
-    @document = Document.find_by_param(params[:document_id])
-    @image = Image.find_by(id: params[:image_id])
-  end
-
   def update
```

Comment:
update and destroy seem very similar to those in document_images controller? Is there a way we can have document_images_controller handle them to avoid the duplication? 

Otherwise feels like we may we want to consider a service class.

Reply:
Indeed they are, but there are two differences that make them difficult to combine - the redirect and the lead_image_id. We needed conditionals and in a couple of instances implemented the workflow for one case but not the other (e.g. for lead images but not other images).

But I agree there's some duplication here, and I'd be up for centralising update/destroy into an ImageService or similar. If that sounds good to you, I can do another commit for it? The ImageUploader service could also be a good thing to refactor into this new service.

Reply:
Sure, could be worth spiking on it to see if that's a nicer way to handle the conditional aspect of it.

On first thought it seemed pretty ripe for a service class since we have about 20 lines of code: https://github.com/alphagov/content-publisher/blob/10890b6cbbc047cbce6951f8fc5fbc8ceb7d2208/app/controllers/document_images_controller.rb#L9-L35 but then actually looking at the code about 3/4 is handling different responses so it might well be that there isn't a lot to gain? What's your thoughts?